### PR TITLE
test for Patch /tasks/self/:id

### PR DESCRIPTION
**Pull request by akshay1502**

### CODE REVIEW

File: test/integration/tasks.test.js [hunk 1]
```diff
@@ -293,4 +293,54 @@ describe('Tasks', function () {
...
+    it('Should return Forbidden error if task is not assigned to self', async function () {
+      const { user: { id, username } } = await userModel.fetchUser({ username: 'akshay' })
+
+      const res = await chai
+        .request(app)
+        .patch(`/tasks/self/${taskId1}`)
+        .set('cookie', `${cookieName}=${authService.generateAuthToken({ userId: id })}`)
+
+      const { taskData: { assignee } } = await tasks.fetchTask(taskId1)
+      if (username !== assignee) {
```
Comment (swarajpure) [inline test/integration/tasks.test.js:340]: Why is this line here? 😅

File: test/integration/tasks.test.js [hunk 2]
```diff
@@ -293,4 +293,51 @@ describe('Tasks', function () {
...
+        .end((err, res) => {
+          if (err) { return done(err) }
+          expect(res).to.have.status(404)
+          expect(res.body.message).to.equal('Task doesn\'t exist')
+          return done()
+        })
+    })
+
+    it('Should return Forbidden error if task is not assigned to self', async function () {
+      const { user: { id } } = await userModel.fetchUser({ username: 'akshay' })
```
Comment (swarajpure) [inline test/integration/tasks.test.js:332]: you'll first need to add this user in DB right?

Comment (akshay1502) [reply test/integration/tasks.test.js:332]: I don't think we need to add user in DB.

File: test/integration/tasks.test.js [hunk 3]
```diff
@@ -293,4 +294,55 @@ describe('Tasks', function () {
...
+          expect(res.body.message).to.equal('Task doesn\'t exist')
+          return done()
+        })
+    })
+
+    it('Should return Forbidden error if task is not assigned to self', async function () {
+      const { userId } = await userModel.addOrUpdate({
+        github_id: 'akshay1502',
+        username: 'akshay'
+      })
```
Comment (swarajpure) [inline test/integration/tasks.test.js:336]: Better to use some user from the fixtures here?
Also please have a look at line 26, the function and the way to use the function

File: test/integration/tasks.test.js [hunk 4]
```diff
@@ -293,4 +294,55 @@ describe('Tasks', function () {
...
+
+      const res = await chai
+        .request(app)
+        .patch(`/tasks/self/${taskId1}`)
+        .set('cookie', `${cookieName}=${authService.generateAuthToken({ userId })}`)
```
Comment (swarajpure) [inline test/integration/tasks.test.js:341]: Please let's keep it a bit clean, we could compute the jwt somewhere else and use it from a variable?

File: test/integration/tasks.test.js [hunk 5]
```diff
@@ -293,4 +294,53 @@ describe('Tasks', function () {
...
+      const jwt = authService.generateAuthToken({ userId })
+
+      const res = await chai
+        .request(app)
+        .patch(`/tasks/self/${taskId1}`)
+        .set('cookie', `${cookieName}=${jwt}`)
+
+      expect(res).to.have.status(403)
+      expect(res.body.message).to.equal('This task is not assigned to you')
+      await cleanDb()
```
Comment (swarajpure) [inline test/integration/tasks.test.js:343]: Can we have this function call inside `after` block  in the TOP level describe block?

File: test/integration/tasks.test.js [hunk 6]
```diff
@@ -293,4 +294,53 @@ describe('Tasks', function () {
         })
     })
   })
+
+  describe('PATCH /self/:id', function () {
```
Comment (swarajpure) [inline test/integration/tasks.test.js:298]: Please add a test for no cookie case, as well

**Review by swarajpure — APPROVED** [review]
Hope you learnt about testing 🎉 
Great job in here! 🔥
